pkg/cloud/rgraph/exec/event.go

```diff
@@ -27,10 +27,32 @@ import (
 type Event interface {
 	// Equal returns true if this event is == to other.
 	Equal(other Event) bool
-	// String implements Stringer.
+	// String implements Stringer. The values returned by String() must satisfy
```

Comment:
This contract is going to be a bit difficult to enforce.  I looked at the current impl, and the `%v` used in the `String()` methods is going to format contained pointers as hex-addresses rather than emitting the contained fields.  Can we update `EventList.Equal` or other code / test locations to defend against implementations that violate this constraint?

Reply:
Not sure how to fix this? Are you suggesting using some form of reflect.DeepEqual

Reply:
Might be able to detect it by keeping a map of `Event.String()` to `Event` then calling `Equal()` on collision.  In any case, the existing events need to get fixed up for this to work.

Reply:
To be specific -- you mean in the EventList Equal() method?